### PR TITLE
chore(Core/Misc): cleanup

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Remove deprecated `unary_function`
- Refactor `ArenaSpectator` (use `fmt:format` style, move to `.cpp` general core, add support dynamic linking)
- Added includes guard
- Rework includes in `Player.h`
- Move player enums and etc to separated file `PlayerCommon.h`

## Issues:
#8281

### CODE REVIEW

File: src/server/game/ArenaSpectator/ArenaSpectator.cpp [hunk 1]
```diff
@@ -133,19 +152,21 @@ namespace ArenaSpectator
         {
             handler->PSendSysMessage("To spectate, please fix the following:");
             for (std::list<std::string>::const_iterator itr = errors.begin(); itr != errors.end(); ++itr)
-                handler->PSendSysMessage("- %s", (*itr).c_str());
+                handler->PSendSysMessage("- {}", (*itr));
```
Comment (Winfidonarleyan) [inline src/server/game/ArenaSpectator/ArenaSpectator.cpp:155]: .

File: src/server/game/Entities/Player/KillRewarder.h [hunk 2]
```diff
@@ -34,3 +43,7 @@ class KillRewarder
     bool _isBattleGround;
     bool _isPvP;
 };
+
+std::ostringstream& operator<< (std::ostringstream& ss, PlayerTaxi const& taxi);
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Player/KillRewarder.h:47]: .

File: src/server/game/Entities/Player/Player.h [hunk 3]
```diff
@@ -7,969 +7,24 @@
...
-struct EntryPointData
-{
-    EntryPointData()
-    {
-        ClearTaxiPath();
-    }
+#include "MapReference.h"
+#include "GroupReference.h"
+#include "Pet.h"
+#include "DbcStores.h"
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Player/Player.h:16]: .

**Comment by FrancescoBorzi** [comment]
what's the status of this PR?

**Comment by Winfidonarleyan** [comment]
> what's the status of this PR?

Need continue

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan maybe better split this in smaller PRs? so they will be easier to be checked and merged

**Comment by Winfidonarleyan** [comment]
> @Winfidonarleyan maybe better split this in smaller PRs? so they will be easier to be checked and merged

This pr just refactor code

**Comment by FrancescoBorzi** [comment]
> > @Winfidonarleyan maybe better split this in smaller PRs? so they will be easier to be checked and merged
> 
> This pr just refactor code

Yes I understand, and it's good :)

I'm just wondering if you can split it into smaller PRs so we can merge them more easily.

**Comment by Winfidonarleyan** [comment]
> > > @Winfidonarleyan maybe better split this in smaller PRs? so they will be easier to be checked and merged
> > 
> > 
> > This pr just refactor code
> 
> Yes I understand, and it's good :)
> 
> I'm just wondering if you can split it into smaller PRs so we can merge them more easily.

Okay, after done this pr, i try

**Comment by acidmanifesto** [comment]
Status of the draft? Is is abandon? or no longer relevant?

**Pull request closed without merging**